src/validation/RegistrationSchema.tsx

```diff
@@ -26,13 +23,54 @@ export const RegistrationSchema = () =>
       description: string().required(
         t`Please let us know why you're interested in Brave Ads`,
       ),
-      marketingChannel: string().required(
-        t`Please let us know how you heard about Brave Ads`,
+    }),
+  });
+
+const SearchRegistrationSchema = () =>
+  object().shape({
+    user: UserSchema(),
+    domain: string()
+      .required(t`Domain is required`)
+      .matches(NoSpacesRegex, t`Domain must not contain any whitespace`)
+      .matches(
+        DomainRegex,
+        t`Please enter a valid domain, for example: brave.com`,
+      )
+      .test(
+        "is-matching-domain",
+        t`Domain must match the email domain`,
+        (value, context) => {
+          const email = context.parent.user.email;
+          if (!email) return false;
+          const splitEmail = email.split("@");
+          return value === splitEmail[splitEmail.length - 1];
+        },
+      ),
```

Comment:
This is pretty complicated validation logic. Please add some tests for this schema.